src/pages/import/MetaData/helper.js

```diff
@@ -0,0 +1,189 @@
+import { useEffect, useState } from 'react'
+import i18n from '@dhis2/d2-i18n'
+
+import {
+    ASYNC_DEFAULT_VALUE,
+    ASYNC_KEY,
+} from '../../../components/Inputs/Async'
+import {
+    ATOMIC_MODE_DEFAULT_VALUE,
+    ATOMIC_MODE_KEY,
+} from '../../../components/Inputs/AtomicMode'
+import {
+    FLUSH_MODE_DEFAULT_VALUE,
+    FLUSH_MODE_KEY,
+} from '../../../components/Inputs/FlushMode'
+import {
+    FORMAT_DEFAULT_VALUE,
+    FORMAT_KEY,
+    OPTION_CSV,
+    OPTION_JSON,
+    OPTION_XML,
+} from '../../../components/Inputs/Format'
+import {
+    IDENTIFIER_DEFAULT_VALUE,
+    IDENTIFIER_KEY,
+} from '../../../components/Inputs/Identifier'
+import {
+    IMPORT_MODE_DEFAULT_VALUE,
+    IMPORT_MODE_KEY,
+} from '../../../components/Inputs/ImportMode'
+import {
+    IMPORT_REPORT_MODE_DEFAULT_VALUE,
+    IMPORT_REPORT_MODE_KEY,
+} from '../../../components/Inputs/ImportReportMode'
+import {
+    IMPORT_STRATEGY_DEFAULT_VALUE,
+    IMPORT_STRATEGY_KEY,
+} from '../../../components/Inputs/ImportStrategy'
+import {
+    INCLUSION_STRATEGY_DEFAULT_VALUE,
+    INCLUSION_STRATEGY_KEY,
+} from '../../../components/Inputs/InclusionStrategy'
+import {
+    MERGE_MODE_DEFAULT_VALUE,
+    MERGE_MODE_KEY,
+} from '../../../components/Inputs/MergeMode'
+import {
+    PREHEAT_MODE_DEFAULT_VALUE,
+    PREHEAT_MODE_KEY,
+} from '../../../components/Inputs/PreheatMode'
+import {
+    SKIP_SHARING_DEFAULT_VALUE,
+    SKIP_SHARING_KEY,
+} from '../../../components/Inputs/SkipSharing'
+import {
+    SKIP_VALIDATION_DEFAULT_VALUE,
+    SKIP_VALIDATION_KEY,
+} from '../../../components/Inputs/SkipValidation'
+import { api, eventEmitter } from '../../../services'
+import { fetchLog } from '../helpers'
+import { getParamsFromFormState } from '../../../helpers/form'
+import { getUploadXHR } from '../../../helpers/xhr'
+import { isProduction } from '../../../helpers/env'
+
+export const supportedFormats = [OPTION_JSON, OPTION_XML, OPTION_CSV]
+
+export const defaultValues = {
+    [FORMAT_KEY]: FORMAT_DEFAULT_VALUE,
+    [IMPORT_MODE_KEY]: IMPORT_MODE_DEFAULT_VALUE,
+    [IDENTIFIER_KEY]: IDENTIFIER_DEFAULT_VALUE,
+    [IMPORT_REPORT_MODE_KEY]: IMPORT_REPORT_MODE_DEFAULT_VALUE,
+    [PREHEAT_MODE_KEY]: PREHEAT_MODE_DEFAULT_VALUE,
+    [IMPORT_STRATEGY_KEY]: IMPORT_STRATEGY_DEFAULT_VALUE,
+    [ATOMIC_MODE_KEY]: ATOMIC_MODE_DEFAULT_VALUE,
+    [MERGE_MODE_KEY]: MERGE_MODE_DEFAULT_VALUE,
+    [FLUSH_MODE_KEY]: FLUSH_MODE_DEFAULT_VALUE,
+    [SKIP_SHARING_KEY]: SKIP_SHARING_DEFAULT_VALUE,
+    [SKIP_VALIDATION_KEY]: SKIP_VALIDATION_DEFAULT_VALUE,
+    [ASYNC_KEY]: ASYNC_DEFAULT_VALUE,
+    [INCLUSION_STRATEGY_KEY]: INCLUSION_STRATEGY_DEFAULT_VALUE,
+}
+
+export const onSubmit = (setLoading, setError) => values => {
+    try {
```

Comment:
Which command are we guarding against here? If possible it'd be nice to wrap it just where things might fail, so it's more clear?

Reply:
I have no idea.. Just copied the existing code. I assume it's in case of a malformed response